File: openapi-generator/csharp-nancyfx/src/Org.OpenAPITools/Models/DocumentCreate.cs
using System;
using System.Collections.Generic;
using System.IO;
using System.Text;
using Sharpility.Extensions;
using NodaTime;

namespace Org.OpenAPITools.v1.Models
{
    /// <summary>
    /// DocumentCreate
    /// </summary>
    public sealed class DocumentCreate:  IEquatable<DocumentCreate>
    { 
        /// <summary>
        /// ID of the Samsara dispatch job for which the document is submitted
        /// </summary>
        public long? DispatchJobId { get; private set; }

        /// <summary>
        /// Notes submitted with this document.
        /// </summary>
        public string Notes { get; private set; }

        /// <summary>
        /// Universally unique identifier for the document type this document is being created for.
        /// </summary>
        public string DocumentTypeUuid { get; private set; }

        /// <summary>
        /// List of fields should match the document type’s list of field types in the correct order. In other words, a field&#39;s valueType and value (i.e. only one of: stringValue, numberValue, or photoValue) at index _i_ should match with the document field type’s valueType at index _i_.
        /// </summary>
        public List<DocumentField> Fields { get; private set; }


        /// <summary>
        /// Empty constructor required by some serializers.
        /// Use DocumentCreate.Builder() for instance creation instead.
        /// </summary>
        [Obsolete]
        public DocumentCreate()
        {
        }

        private DocumentCreate(long? DispatchJobId, string Notes, string DocumentTypeUuid, List<DocumentField> Fields)
        {
            
            this.DispatchJobId = DispatchJobId;
            
            this.Notes = Notes;
            
            this.DocumentTypeUuid = DocumentTypeUuid;
            
            this.Fields = Fields;
            
        }

        /// <summary>
        /// Returns builder of DocumentCreate.
        /// </summary>
        /// <returns>DocumentCreateBuilder</returns>
        public static DocumentCreateBuilder Builder()
        {
            return new DocumentCreateBuilder();
        }

        /// <summary>
        /// Returns DocumentCreateBuilder with properties set.
        /// Use it to change properties.
        /// </summary>
        /// <returns>DocumentCreateBuilder</returns>
        public DocumentCreateBuilder With()
        {
            return Builder()
                .DispatchJobId(DispatchJobId)
                .Notes(Notes)
                .DocumentTypeUuid(DocumentTypeUuid)
                .Fields(Fields);
        }

        public override string ToString()
        {
            return this.PropertiesToString();
        }

        public override bool Equals(object obj)
        {
            return this.EqualsByProperties(obj);
        }

        public bool Equals(DocumentCreate other)
        {
            return Equals((object) other);
        }

        public override int GetHashCode()
        {
            return this.PropertiesHash();
        }

        /// <summary>
        /// Implementation of == operator for (DocumentCreate.
        /// </summary>
        /// <param name="left">Compared (DocumentCreate</param>
        /// <param name="right">Compared (DocumentCreate</param>
        /// <returns>true if compared items are equals, false otherwise</returns>
        public static bool operator == (DocumentCreate left, DocumentCreate right)
        {
            return Equals(left, right);
        }

        /// <summary>
        /// Implementation of != operator for (DocumentCreate.
        /// </summary>
        /// <param name="left">Compared (DocumentCreate</param>
        /// <param name="right">Compared (DocumentCreate</param>
        /// <returns>true if compared items are not equals, false otherwise</returns>
        public static bool operator != (DocumentCreate left, DocumentCreate right)
        {
            return !Equals(left, right);
        }

        /// <summary>
        /// Builder of DocumentCreate.
        /// </summary>
        public sealed class DocumentCreateBuilder
        {
            private long? _DispatchJobId;
            private string _Notes;
            private string _DocumentTypeUuid;
            private List<DocumentField> _Fields;

            internal DocumentCreateBuilder()
            {
                SetupDefaults();
            }

            private void SetupDefaults()
            {
            }

            /// <summary>
            /// Sets value for DocumentCreate.DispatchJobId property.
            /// </summary>
            /// <param name="value">ID of the Samsara dispatch job for which the document is submitted</param>
            public DocumentCreateBuilder DispatchJobId(long? value)
            {
                _DispatchJobId = value;
                return this;
            }

            /// <summary>
            /// Sets value for DocumentCreate.Notes property.
            /// </summary>
            /// <param name="value">Notes submitted with this document.</param>
            public DocumentCreateBuilder Notes(string value)
            {
                _Notes = value;
                return this;
            }

            /// <summary>
            /// Sets value for DocumentCreate.DocumentTypeUuid property.
            /// </summary>
            /// <param name="value">Universally unique identifier for the document type this document is being created for.</param>
            public DocumentCreateBuilder DocumentTypeUuid(string value)
            {
                _DocumentTypeUuid = value;
                return this;
            }

            /// <summary>
            /// Sets value for DocumentCreate.Fields property.
            /// </summary>
            /// <param name="value">List of fields should match the document type’s list of field types in the correct order. In other words, a field&#39;s valueType and value (i.e. only one of: stringValue, numberValue, or photoValue) at index _i_ should match with the document field type’s valueType at index _i_.</param>
            public DocumentCreateBuilder Fields(List<DocumentField> value)
            {
                _Fields = value;
                return this;
            }


            /// <summary>
            /// Builds instance of DocumentCreate.
            /// </summary>
            /// <returns>DocumentCreate</returns>
            public DocumentCreate Build()
            {
                Validate();
                return new DocumentCreate(
                    DispatchJobId: _DispatchJobId,
                    Notes: _Notes,
                    DocumentTypeUuid: _DocumentTypeUuid,
                    Fields: _Fields
                );
            }

            private void Validate()
            { 
                if (_DocumentTypeUuid == null)
                {
                    throw new ArgumentException("DocumentTypeUuid is a required property for DocumentCreate and cannot be null");
                } 
                if (_Fields == null)
                {
                    throw new ArgumentException("Fields is a required property for DocumentCreate and cannot be null");
                } 
            }
        }

        
    }
}

File: openapi-generator/java-pkmst/src/main/java/com/prokarma/pkmst/model/DocumentCreate.java
package com.prokarma.pkmst.model;

import java.util.Objects;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonCreator;
import com.prokarma.pkmst.model.DocumentBase;
import com.prokarma.pkmst.model.DocumentField;
import io.swagger.annotations.ApiModel;
import io.swagger.annotations.ApiModelProperty;
import java.util.ArrayList;
import java.util.List;
/**
 * Response class to be returned by Api
 * @author pkmst
 *
 */
/**
 * DocumentCreate
 */

@javax.annotation.Generated(value = "org.openapitools.codegen.languages.JavaPKMSTServerCodegen", date = "2019-03-01T05:34:34.656Z[GMT]")

public class DocumentCreate   {
  @JsonProperty("dispatchJobId")
  private Long dispatchJobId;

  @JsonProperty("notes")
  private String notes;

  @JsonProperty("documentTypeUuid")
  private String documentTypeUuid;

  @JsonProperty("fields")
  
  private List<DocumentField> fields = new ArrayList<>();

  public DocumentCreate dispatchJobId(Long dispatchJobId) {
    this.dispatchJobId = dispatchJobId;
    return this;
  }

   /**
   * ID of the Samsara dispatch job for which the document is submitted
   * @return dispatchJobId
  **/
  @ApiModelProperty(example = "773", value = "ID of the Samsara dispatch job for which the document is submitted")
  public Long getDispatchJobId() {
    return dispatchJobId;
  }

  public void setDispatchJobId(Long dispatchJobId) {
    this.dispatchJobId = dispatchJobId;
  }

  public DocumentCreate notes(String notes) {
    this.notes = notes;
    return this;
  }

   /**
   * Notes submitted with this document.
   * @return notes
  **/
  @ApiModelProperty(example = "Fueled up before delivery.", value = "Notes submitted with this document.")
  public String getNotes() {
    return notes;
  }

  public void setNotes(String notes) {
    this.notes = notes;
  }

  public DocumentCreate documentTypeUuid(String documentTypeUuid) {
    this.documentTypeUuid = documentTypeUuid;
    return this;
  }

   /**
   * Universally unique identifier for the document type this document is being created for.
   * @return documentTypeUuid
  **/
  @ApiModelProperty(example = "4aff772c-a7bb-45e6-8e41-6a53e34feb83", required = true, value = "Universally unique identifier for the document type this document is being created for.")
  public String getDocumentTypeUuid() {
    return documentTypeUuid;
  }

  public void setDocumentTypeUuid(String documentTypeUuid) {
    this.documentTypeUuid = documentTypeUuid;
  }

  public DocumentCreate fields(List<DocumentField> fields) {
    this.fields = fields;
    return this;
  }

  public DocumentCreate addFieldsItem(DocumentField fieldsItem) {
    this.fields.add(fieldsItem);
    return this;
  }

   /**
   * List of fields should match the document type’s list of field types in the correct order. In other words, a field's valueType and value (i.e. only one of: stringValue, numberValue, or photoValue) at index _i_ should match with the document field type’s valueType at index _i_.
   * @return fields
  **/
  @ApiModelProperty(example = "[{label=Driver Photo, photoValue=[], valueType=ValueType_Photo}, {label=Gallons, numberValue=12.34, valueType=ValueType_Number}, {label=City, stringValue=San Francisco, CA, valueType=ValueType_String}]", required = true, value = "List of fields should match the document type’s list of field types in the correct order. In other words, a field's valueType and value (i.e. only one of: stringValue, numberValue, or photoValue) at index _i_ should match with the document field type’s valueType at index _i_.")
  public List<DocumentField> getFields() {
    return fields;
  }

  public void setFields(List<DocumentField> fields) {
    this.fields = fields;
  }


  @Override
  public boolean equals(java.lang.Object o) {
    if (this == o) {
      return true;
    }
    if (o == null || getClass() != o.getClass()) {
      return false;
    }
    DocumentCreate documentCreate = (DocumentCreate) o;
    return Objects.equals(this.dispatchJobId, documentCreate.dispatchJobId) &&
        Objects.equals(this.notes, documentCreate.notes) &&
        Objects.equals(this.documentTypeUuid, documentCreate.documentTypeUuid) &&
        Objects.equals(this.fields, documentCreate.fields);
  }

  @Override
  public int hashCode() {
    return Objects.hash(dispatchJobId, notes, documentTypeUuid, fields);
  }

  @Override
  public String toString() {
    StringBuilder sb = new StringBuilder();
    sb.append("class DocumentCreate {\n");
    
    sb.append("    dispatchJobId: ").append(toIndentedString(dispatchJobId)).append("\n");
    sb.append("    notes: ").append(toIndentedString(notes)).append("\n");
    sb.append("    documentTypeUuid: ").append(toIndentedString(documentTypeUuid)).append("\n");
    sb.append("    fields: ").append(toIndentedString(fields)).append("\n");
    sb.append("}");
    return sb.toString();
  }

  /**
   * Convert the given object to string with each line indented by 4 spaces
   * (except the first line).
   */
  private String toIndentedString(java.lang.Object o) {
    if (o == null) {
      return "null";
    }
    return o.toString().replace("\n", "\n    ");
  }
}

